drivers/gpio/Kconfig

```diff
@@ -93,6 +93,14 @@ config GPIO_ENABLE_DISABLE_INTERRUPT
 	  on/off the interrupt signal without changing other registers, such as
 	  pending register, etc. The driver must implement it to work.
 
+config GPIO_DIAGNOSTICS
+	bool "Enable advanced diagnostic functionality"
+	help
+	  This option enables diagnostic support for Maxim industrial input/output
+	  chips which have advance functionality for wire break, short circuit,
+	  over/under voltage and other. In mater diagnostic to work it should be
+	  in driver in self on per IC bases.
+
```

Comment:
We cannot have what looks like a vendor-specific API extension available in the top-level GPIO driver subsystem.

Reply:
Ok I understand, make sens. My intention was to have something generic , not vendor specific. I will remove Maxim part.